File: test/aidbox_sdk/snapshots/typescript/hl7-fhir-r4-core/Procedure.ts
import { Annotation } from "./Annotation";
import { Age } from "./Age";
import { Period } from "./Period";
import { CodeableConcept } from "./CodeableConcept";
import { Range } from "./Range";
import { DomainResource } from "./DomainResource";
import { Reference } from "./Reference";
import { Identifier } from "./Identifier";
import { BackboneElement } from "./BackboneElement";

export type ProcedureFocalDevice = BackboneElement & {
    action?: CodeableConcept;
    manipulated: Reference;
};

export type ProcedurePerformer = BackboneElement & {
    actor: Reference;
    function?: CodeableConcept;
    onBehalfOf?: Reference;
};

export type Procedure = DomainResource & {
    category?: CodeableConcept;
    report?: Reference[];
    usedCode?: CodeableConcept[];
    usedReference?: Reference[];
    instantiatesCanonical?: string[];
    instantiatesUri?: string[];
    focalDevice?: ProcedureFocalDevice[];
    encounter?: Reference;
    complicationDetail?: Reference[];
    reasonCode?: CodeableConcept[];
    statusReason?: CodeableConcept;
    performed?: Age | string | Range | Period;
    outcome?: CodeableConcept;
    asserter?: Reference;
    note?: Annotation[];
    complication?: CodeableConcept[];
    status: string;
    recorder?: Reference;
    code?: CodeableConcept;
    identifier?: Identifier[];
    bodySite?: CodeableConcept[];
    basedOn?: Reference[];
    partOf?: Reference[];
    location?: Reference;
    followUp?: CodeableConcept[];
    subject: Reference;
    performer?: ProcedurePerformer[];
    reasonReference?: Reference[];
};

File: test/aidbox_sdk/snapshots/python/hl7-fhir-r4-core/Procedure.py
from typing import Optional, List
from pydantic import *
from ..base.Annotation import Annotation
from ..base.Age import Age
from ..base.Period import Period
from ..base.CodeableConcept import CodeableConcept
from ..base.Range import Range
from ..base.DomainResource import DomainResource
from ..base.Reference import Reference
from ..base.Identifier import Identifier
from ..base.BackboneElement import BackboneElement

class Procedure_FocalDevice(BackboneElement):
    action: Optional[CodeableConcept] = None
    manipulated: Reference

class Procedure_Performer(BackboneElement):
    actor: Reference
    function: Optional[CodeableConcept] = None
    on_behalf_of: Optional[Reference] = None

class Procedure(DomainResource):
    category: Optional[CodeableConcept] = None
    report: Optional[List[Reference]] = None
    used_code: Optional[List[CodeableConcept]] = None
    used_reference: Optional[List[Reference]] = None
    instantiates_canonical: Optional[List[str]] = None
    instantiates_uri: Optional[List[str]] = None
    focal_device: Optional[List[Procedure_FocalDevice]] = None
    encounter: Optional[Reference] = None
    performed_age: Optional[Age] = None
    complication_detail: Optional[List[Reference]] = None
    reason_code: Optional[List[CodeableConcept]] = None
    performed_string: Optional[str] = None
    status_reason: Optional[CodeableConcept] = None
    outcome: Optional[CodeableConcept] = None
    asserter: Optional[Reference] = None
    note: Optional[List[Annotation]] = None
    performed_range: Optional[Range] = None
    complication: Optional[List[CodeableConcept]] = None
    status: str
    performed_date_time: Optional[str] = None
    recorder: Optional[Reference] = None
    code: Optional[CodeableConcept] = None
    identifier: Optional[List[Identifier]] = None
    body_site: Optional[List[CodeableConcept]] = None
    based_on: Optional[List[Reference]] = None
    part_of: Optional[List[Reference]] = None
    performed_period: Optional[Period] = None
    location: Optional[Reference] = None
    follow_up: Optional[List[CodeableConcept]] = None
    subject: Reference
    performer: Optional[List[Procedure_Performer]] = None
    reason_reference: Optional[List[Reference]] = None